generator_process/actions/prompt_to_image.py

```diff
@@ -10,99 +10,86 @@
 import random
 from .detect_seamless import SeamlessAxes
 
+from ..models import Pipeline
 
-class Pipeline(enum.IntEnum):
-    STABLE_DIFFUSION = 0
-
-    STABILITY_SDK = 1
-
-    @staticmethod
-    def local_available():
-        from ...absolute_path import absolute_path
-        return os.path.exists(absolute_path(".python_dependencies/diffusers"))
-
-    def __str__(self):
-        return self.name
+def load_pipe(self, generator_pipeline, model, optimizations, scheduler, device):
```

Comment:
This will wrongly use the cached pipe when switching between prompt to image and image to image. Would also be improved by recreating the pipe for changes in device and half precision.
```suggestion
def load_pipe(self, action, generator_pipeline, model, optimizations, scheduler, device):
```